lib/vsc/install/shared_setup.py

```diff
@@ -1254,13 +1275,17 @@ def parse_target(self, target, urltemplate=None):
 
         # update the cmdclass with ones from vsc_setup_klass
         # cannot do this in one go, when SHARED_TARGET is defined, vsc_setup doesn't exist yet
-        for name, klass in new_target['cmdclass'].items():
+        keepers = new_target.copy()
+        for name in list(new_target['cmdclass']):
+            klass = new_target['cmdclass'][name]
             try:
-                new_target['cmdclass'][name] = getattr(vsc_setup_klass, klass.__name__)
+                keepers['cmdclass'][name] = getattr(vsc_setup_klass, klass.__name__)
             except AttributeError:
-                del new_target['cmdclass'][name]
+                del keepers['cmdclass'][name]
+                log.info("Not including new_target['cmdclass']['%s']" %name)
```

Comment:
style nitpicking: missing space before `name`...

Reply:
where is the pep8 robot when you need him?!

Reply:
Yeah, maybe we should enable https://houndci.com/ here too, it has worked out quite well in the EasyBuild repos...